plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/worker/SerializableReporterProviderTest.kt

```diff
@@ -0,0 +1,47 @@
+package org.jlleitschuh.gradle.ktlint.worker
+
+import com.pinterest.ktlint.core.Reporter
+import com.pinterest.ktlint.core.ReporterProvider
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.io.PrintStream
+
+internal class SerializableReporterProviderTest {
+    @TempDir
+    lateinit var temporaryFolder: File
+
+    @Test
+    internal fun `Should correctly serialize and deserialize ReporterProvider`() {
```

Comment:
Don't feel obligated to mark test classes/methods as `internal`. The default access modifiers are completely fine.